helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -533,7 +533,7 @@ public void invoke(NotificationContext changeContext) throws Exception {
     }
   }
 
-  private void subscribeChildChange(String path, NotificationContext.Type callbackType) {
+  private List<String> subscribeChildChange(String path, NotificationContext.Type callbackType) {
     if (callbackType == NotificationContext.Type.INIT
```

Comment:
I feel like we could create an empty list here. So we won’t return a null and no need to do the nullptr check later.

Reply:
I thought about it. There is still a possibility that `childrenSubscribeResult.getChildren()` returns `null`. The caller still needs to do the `null` check. So I just make it consistent with `zkClient.subscribeChildChanges` and let the caller determine the behavior.

I also thought about using `Optional`, but it doesn't seem that would add too much value. And `Optional` will add more overhead than `null`. The original code also does null check, so for now I'd keep the change minimum. Later we will still have a better design and hopefully will make the code cleaner.

Reply:
Is it possible that you just return ChildrenSubscribeResult?
Then if _isInstalled is false, no one will read the children. Otherwise, if _isInstalled, then the list should never be null.

Reply:
I also thought about it. But from the code, if `_isInstalled == true`, it is still possible that the children list is null: 
```
        try {
          return getChildren(path, true);
        } catch (ZkNoNodeException e) {
          // ignore, the "exists" watch will listen for the parent node to appear
          LOG.info("watchForChilds path not existing:{} skipWatchingNodeNoteExist: {}",
              path, skipWatchingNonExistNode);
        }
        return null;
```
So we could not determine if children list is null or not based on `_isInstalled`.